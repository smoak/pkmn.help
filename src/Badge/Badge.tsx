import React, { FC } from "react";
import classnames from "classnames";

import { Type } from "../data";

export interface BadgeProps {
  readonly pokemonType: Type;
}

const Badge: FC<BadgeProps> = ({ pokemonType }) => {
  const style = { minWidth: "7.5em" };
  const className = classnames(
    `type-${pokemonType}`,
    "ba b--black-10",
    "badge",
    "with-border-color",
    "dib pv2 ph3",
    "br1",
    "ma--2px",
    "ttu tc b f5 f4-l"
  );

  return (
    <div key={pokemonType} className={className} style={style}>
      {pokemonType}
    </div>
  );
};

export default Badge;
