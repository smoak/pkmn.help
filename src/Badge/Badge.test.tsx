import React from "react";
import { ShallowWrapper, shallow } from "enzyme";

import Badge from "./Badge";
import { Type } from "../data";

describe("Badge", () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<Badge pokemonType={Type.NORMAL} />);
  });

  it("renders", () => {
    expect(component).toMatchSnapshot();
  });
});
