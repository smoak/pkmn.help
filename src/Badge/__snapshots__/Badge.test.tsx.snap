// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders 1`] = `
<div
  className="type-normal ba b--black-10 badge with-border-color dib pv2 ph3 br1 ma--2px ttu tc b f5 f4-l"
  key="normal"
  style={
    Object {
      "minWidth": "7.5em",
    }
  }
>
  normal
</div>
`;
